### PR TITLE
mitm domains

### PR DESCRIPTION
**Pull request by mmatczuk**

### CODE REVIEW

File: cmd/forwarder/run/run.go [hunk 1]
```diff
@@ -154,8 +155,19 @@ func (c *command) runE(cmd *cobra.Command, _ []string) (cmdErr error) {
 	if c.mitmConfig.CACertFile != "" {
 		c.mitm = true
 	}
+	if len(c.mitmDomains) > 0 {
+		c.mitm = true
+	}
```
Comment (Choraden) [inline cmd/forwarder/run/run.go:160]: How about we 'or' it with the condition above?

Comment (mmatczuk) [reply cmd/forwarder/run/run.go]: Fixed that.

File: e2e/tests/flag_test.go [hunk 2]
```diff
@@ -117,12 +117,19 @@ func TestFlagMITMDomains(t *testing.T) {
...
+func TestFlagDenyDomains(t *testing.T) {
+	t.Run("include(google)", func(t *testing.T) {
+		newClient(t, "https://www.google.com").GET("/").
+			ExpectStatus(http.StatusForbidden)
+	})
+
+	t.Run("exclude (httpbin)", func(t *testing.T) {
+		newClient(t, httpbin).GET("/status/200").
+			ExpectStatus(http.StatusOK)
+	})
```
Comment (Choraden) [inline e2e/tests/flag_test.go:129]: This change does not resonate with the commit name. 
Also make subtests names consistent - spaces.

**Comment by Choraden** [comment]
LGTM. 

Good job with [martian (mitm): use Peek and avoid allocating reader for 1 byte slice](https://github.com/saucelabs/forwarder/pull/434/commits/45cb6201305b35f116f69de6db1c5dfebf0fe320) I really like this change.

File: e2e/tests/flag_test.go (hunk 2 above)
Comment (mmatczuk) [reply e2e/tests/flag_test.go:129]: Spaces makes it look like `foo_(` vs `foo(`

Comment (mmatczuk) [reply e2e/tests/flag_test.go:129]: Updated the commit msg

**Comment by mmatczuk** [comment]
Fixes #336

**Pull request merged**